src/datasets/packaged_modules/generator/generator.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+from typing import Callable, Optional
+
+import datasets
+
+
+@dataclass
+class GeneratorConfig(datasets.BuilderConfig):
+    generator: Optional[Callable] = None
```

Comment:
Maybe you don't need to set None by default ?

Reply:
I would expect this to work, but it doesn't due to the class inheritance (see https://bugs.python.org/issue39300 for the explanation)